src/components/modules/caret.ts

```diff
@@ -591,4 +685,73 @@ export default class Caret extends Module {
 
     return siblings;
   }
+
+  /**
+   * Judge if next or previous line is existed
+   *
+   * @param {boolean} isNext - if true, search next line. Previous otherwise
+   */
+  private isLineExisted(isNext: boolean): boolean {
+    const { BlockManager } = this.Editor;
+    const range = new Range();
+
+    const currentBoundingClientRect = Selection.get().getRangeAt(0)
+      .getBoundingClientRect();
+
+    const isBroken = this.walkTextNodeChars(BlockManager.currentBlock.currentInput, isNext, (textNode, index) => {
+      range.setStart(textNode, index);
+      range.setEnd(textNode, index + 1);
+
+      const boundingClientRect = range.getBoundingClientRect();
+
+      if (isNext && currentBoundingClientRect.y < boundingClientRect.y) {
```

Comment:
please add comments for these parts of the code

Reply:
Fixed.